Common/util/file.cpp

```diff
@@ -250,103 +249,130 @@ Stream *File::OpenStderr()
     return FileStream::WrapHandle(stderr, kFile_Write, kDefaultSystemEndianess);
 }
 
-String File::FindFileCI(const String &dir_name, const String &file_name)
+String File::FindFileCI(const String &base_dir, const String &file_name,
+    bool is_dir, String *most_found, String *not_found)
 {
 #if !defined (AGS_CASE_SENSITIVE_FILESYSTEM)
     // Simply concat dir and filename paths
-    return Path::ConcatPaths(dir_name, file_name);
+    return Path::ConcatPaths(base_dir, file_name);
```

Comment:
Shouldn't we be checking if the file exists here instead of returning a path?

Reply:
Is this what you anted to do here?
```
    String path = Path::ConcatPaths(base_dir, file_name);
    if (ags_file_exists(path.GetCStr()) != 0)
        return path;
    return {};
```

Reply:
No, this function is not supposed to check whether file exists, unless there's no other way (for case insensitive filesystems).

Reply:
But yes, File_Exists is broken, because previously it was testing for file existence, now it incorrectly assumes that it's already tested by the call to ResolveScriptPathAndFindFile.

Reply:
It's possible that FindFileCI will need to be changed for case insensitive filesystems to also test for a file's presence, as you suggested, to make things consistent. But then all the uses of this function should be revisited, and possibly adjusted. From a quick glance, not all of them assume this tests for a file presence, or may fail.

Reply:
@AlanDrake pushed a fix: 8e67e325ebbac58a5cad4ba85ca620fce7fef4bf